### DISCUSSION TITLE
error when I run "php artisan serve" command

### DISCUSSION THREAD

**Discussion by inaciowagner** [laravel/framework#45114] [Q&A]
when I run this command **"php artisan serve --host=10.0.0.62 --port=8000"** 
I recive this error like return:
**"PHP Warning:  require(/var/www/html/curso-laravel/vendor/autoload.php): Failed to open stream: No such file or directory in /var/www/html/curso-laravel/artisan on line 18
PHP Fatal error:  Uncaught Error: Failed opening required '/var/www/html/curso-laravel/vendor/autoload.php' (include_path='.:/usr/share/php') in /var/www/html/curso-laravel/artisan:18
Stack trace:
#0 {main}
  thrown in /var/www/html/curso-laravel/artisan on line 18"**
  
  Somebody, please, help me!

**Comment by henzeb** [comment]
run `composer install` first?

**Comment by alfakiat** [comment]
This error throw when you missed `vendor/` directory.
to solve it you should install a missing dependency as follow:

regenerate the `vendor/` folder and the `autoload.php`
```
composer update
```

then execute this command to install required vendors
```
composer install
```

**Reply by henzeb** [reply]
you don't need the update statement at all. this would also update the composer.lock to reflect the latest versions of dependencies. Which might not what the user wants. Composer install also creates the vendor folder.

**Reply by alfakiat** [reply]
@henzeb you are right.
But this error throw when you clone a project and dont install vendors.
 And you maybe would to be up-to-date with latest versions.

**Comment by semicolonn** [comment]
I have same problem to all my downloaded projects of laravel from github.

**Comment by semicolonn** [comment]
ed dependencies. It is recommended that you run `composer update` or `composer update <package name>`.      
Your lock file does not contain a compatible set of packages. Please run composer update.

  Problem 1
    - simplesoftwareio/simple-qrcode is locked to version 4.2.0 and an update of this package was not requested.
    - simplesoftwareio/simple-qrcode 4.2.0 requires ext-gd * -> it is missing from your system. Install or enable PHP's gd extension.

To enable extensions, verify that they are enabled in your .ini files:
    - C:\xampp\php\php.ini
You can also run `php --ini` in a terminal to see which files are used by PHP in CLI mode.
Alternatively, you can run Composer with `--ignore-platform-req=ext-gd` to temporarily ignore these required extensions.

**Reply by myusuf591** [reply]
not run sir in amazon linux2
